FirebaseSessions/Tests/Unit/Error+EquatableTests.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import FirebaseSessions
+
+/// This file and extension exist for ease of testing. Without this, you cannot use
+/// XCTAssertEqual on errors, which means you need to do switches whenever you
+/// want to assert on errors, which is clunky.
+///
+/// This class exists for unit testing purposes only. The SDK should use switch internally
+/// when handling errors, because equating errors is prone to issues (eg. we're just comparing
+/// the types, but not the values).
+extension FirebaseSessionsError: Equatable {
+  public static func == (lhs: FirebaseSessions.FirebaseSessionsError,
+                         rhs: FirebaseSessions.FirebaseSessionsError) -> Bool {
+    return String(reflecting: lhs) == String(reflecting: rhs)
+  }
+}
+
+enum FakeError: Error {
+  case Fake
+}
+
+final class ErrorEquatableTests: XCTestCase {
+  func test_equalErrorTypes_areEqual() throws {
```

Comment:
Is the whole point of these 2 tests to verify the extension above? and the extension above is used in other tests?

Reply:
Exactly. I want to use XCTAssertEqual on Errors, but Errors do not have equality checks natively, and I didn't want to put Error equality into the SDK because it's not perfect.

Let me add a comment on that

Reply:
Oh nvm I have a comment at the top of the file